src/read.jl

```diff
@@ -104,113 +104,160 @@ function read_storage(filepath)::Storage
+            * "to occur."
+        )
+        demand_flexibility["enabled"] = "not_specified"
     end
 
-    # Set the demand flexibility costs and parameters
-    if demand_flexibility["enabled"]
-        # Pre-specify the demand flexibility parameters
-        demand_flexibility["duration"] = nothing
-        demand_flexibility["interval_balance"] = true
-        demand_flexibility["rolling_balance"] = true
-
-        # Try loading the demand flexibility parameters
-        demand_flexibility_parameters = DataFrames.DataFrame()
-        try
-            demand_flexibility_parameters = CSV.File(
-                joinpath(filepath, "demand_flexibility_parameters.csv")
-            ) |> DataFrames.DataFrame
-            println("...loading demand flexibility parameters")
-
-            # Create a dictionary to hold the error messages relevant to loading the 
-                # demand flexibility parameters
-            demand_flexibility_params_errs = Dict()
-            demand_flexibility_params_errs["duration"] = (
-                "The demand flexibility duration parameter is not defined. Will "
-                * "default to nothing."
-            )
-            demand_flexibility_params_errs["enabled"] = (
-                "The parameter that indicates if demand flexibility is enabled is not "
-                * "defined. Will default to being enabled."
-            )
-            demand_flexibility_params_errs["interval_balance"] = (
-                "The parameter that indicates if the interval load balance constraint "
-                * "is enabled is not defined. Will default to being enabled."
-            )
-            demand_flexibility_params_errs["rolling_balance"] = (
-                "The parameter that indicates if the rolling load balance constraint "
-                * "is enabled is not defined. Will default to being enabled."
-            )
+    # Check the feasibility of the duration parameter
+    if demand_flexibility["duration"] == nothing
+        demand_flexibility["duration"] = interval
+    elseif demand_flexibility["duration"] > interval
+        @warn (
+            "Demand flexibility durations greater than the interval length are "
+            * "set equal to the interval length."
+        )
+        demand_flexibility["duration"] = interval
+    end
 
-            # Try assigning the different demand flexibility parameters from the file
-            for k in keys(demand_flexibility_params_errs)
-                try
-                    demand_flexibility[k] = demand_flexibility_parameters[1, k]
-                catch e
-                    println(demand_flexibility_params_errs[k])
-                end
-            end
+    # Prevent the rolling_balance constraint according to the duration parameter
+    if demand_flexibility["rolling_balance"]
+        if demand_flexibility["duration"] == interval
+            demand_flexibility["rolling_balance"] = false
+        end
+    end
```

Comment:
Would this work `demand_flexibility["rolling_balance"] &= ~(demand_flexibility["duration"]==interval)`?

Reply:
Hmm not sure why I opted for the nested `if` statements in the first place. This is definitely much snazzier, thanks!

Reply:
Done.